app/components/Form/ToggleSwitch.tsx

```diff
@@ -0,0 +1,84 @@
+import { Flex } from '@webkom/lego-bricks';
+import cx from 'classnames';
+import { Keyboard } from 'app/utils/constants';
+import { createField } from './Field';
+import styles from './ToggleSwitch.module.css';
+import type {
+  ComponentProps,
+  InputHTMLAttributes,
+  KeyboardEvent,
+  MouseEvent,
+} from 'react';
+import type { Overwrite } from 'utility-types';
+
+type Props = {
+  className?: string;
+} & Overwrite<
+  InputHTMLAttributes<HTMLInputElement>,
+  { onChange?: (checked: boolean) => void }
+>;
+
+const ToggleSwitch = ({
+  id,
+  checked,
+  value,
+  className,
+  disabled,
+  ...props
+}: Props) => {
+  const handleKeyDown = (event: KeyboardEvent<HTMLInputElement>) => {
+    if (event.key === Keyboard.ENTER) {
+      event.preventDefault();
+      const inputElement = event.target as HTMLInputElement;
+      inputElement.click();
+    }
+  };
+
+  const handleClick = (e: MouseEvent<HTMLInputElement>) => {
+    e.preventDefault();
+    if (!disabled) {
+      props.onChange?.(!isChecked);
+    }
+  };
+
+  const isInFormContext = value !== undefined;
+  // Supporting both string and boolean values
+  const isChecked = isInFormContext
+    ? typeof value === 'string'
+      ? value === 'true'
+      : !!value
+    : typeof checked === 'string'
+      ? checked === 'true'
+      : !!checked;
+
+  return (
+    <Flex wrap alignItems="center" gap="var(--spacing-sm)">
+      <div className={cx(styles.toggleSwitch, className)} onClick={handleClick}>
+        <input
+          {...props}
+          type="checkbox"
+          role="switch"
```

Comment:
```suggestion
          role="switch"
          aria-checked={isChecked}
```
The `aria-checked` attribute is required when using the "switch" role. https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/switch_role#associated_aria_roles_states_and_properties

Reply:
Added!